exonum/src/blockchain/tests.rs

```diff
@@ -402,6 +506,40 @@ mod memorydb_tests {
         let mut blockchain = create_blockchain(path);
         super::handling_tx_panic_storage_error(&mut blockchain);
     }
+
+    #[test]
+    fn test_service_execute() {
+        let dir = TempDir::new(super::gen_tempdir_name().as_str()).unwrap();
```

Comment:
I think it's better to move the body of these tests into a separate function.

Reply:
Temp directory destroys when `dir` out of scope.

Reply:
You can return it from init function

Reply:
Yes, I can. But what is a profit ?
You propose change 
`let dir = TempDir::new(super::gen_tempdir_name().as_str()).unwrap();`
for
```rust
fn create_temp_dir() -> TempDir {
    TempDir::new(super::gen_tempdir_name().as_str()).unwrap()
}
...
let dir = create_temp_dir();
```
???